binder/src/main/java/io/grpc/binder/internal/BinderTransportSecurity.java

```diff
@@ -51,8 +57,10 @@ private BinderTransportSecurity() {}
    * @param serverBuilder The ServerBuilder being used to create the server.
    */
   @Internal
-  public static void installAuthInterceptor(ServerBuilder<?> serverBuilder) {
-    serverBuilder.intercept(new ServerAuthInterceptor());
+  public static void installAuthInterceptor(
```

Comment:
FWIW, to manage the executorPool lifetime, we could have this method return a Closeable which the interceptor implements, and which the BinderServer instance calls when it's shutdown.

Then the interceptor could release the executor when the last call completes after a call to close().

Reply:
I started doing this but got stuck in the part about

> [Closeable] which the BinderServer instance calls when it's shutdown

The interceptor is installed in [BinderServerBuilder.build()](https://github.com/grpc/grpc-java/blob/bdb0c72d45a5bbadfc3125ae639c256f7b6ea9a5/binder/src/main/java/io/grpc/binder/BinderServerBuilder.java#L174). IIUC, I would need to pass the new `Closeable`to `ServerImplBuilder` and then to `ServerImpl`. That starts to touch non-binder code and `ServerImpl` currently doesn't seem to track any closeable that is closed during shutdown.

Is this what you had in mind? It's starting to change more classes than I intended.

Reply:
I see. I think I got it right now, PTAL.

I also pulled up the executor lifecycle from `BinderTransportSecurity.installAuthInterceptor` to `BinderServerBuilder.build()`. It felt a bit weird to have another method (`installAuthInterceptor`) return a `Closeable` without any visibility into what kind of resource it holds (though that may just be my opinion).

The last thing I dislike is the fact that `Closeable.close()` throws `IOException` that we know can't really happen. Treating it as http://go/java-practices/impossible#handle for now, but we could also define our own non-throwing "Closeable" interface if we want to make it clearer that no exception is expected (uncertain if this would be overengineered...).